File: python/src/algorithms/insertion.py
from .base import SortAlgorithm


class InsertionSort(SortAlgorithm):
    name = "InsertionSort"

    def sort(self, data, m):
        for i in range(1, len(data)):
            key = data[i]
            j = i
            while j >= 0:
                m.comparisons += 1
                if key < data[j - 1]:
                    break

                data[j] = data[j - 1]
                j -= 1
                m.swaps += 1
            data[j] = key

File: rust/src/algorithms/insertion.rs
use super::base::SortAlgorithm;

pub struct InsertionSort;

impl SortAlgorithm for InsertionSort {
    fn name(&self) -> &'static str {
        "InsertionSort"
    }

    fn sort(&self, data: &mut [i32]) {
        // todo!()
    }
}